src/litserve/loops/base.py

```diff
@@ -250,23 +250,19 @@ def populate_context(self, lit_spec: LitSpec, request: Any):
     def put_response(
         self, response_queues: List[Queue], response_queue_id: int, uid: str, response_data: Any, status: LitAPIStatus
     ) -> None:
-        if self.socket:
-            self.socket.send_pyobj((uid, (response_data, status)))
+        if self.producer:
+            self.producer.put((uid, (response_data, status)), consumer_id=response_queue_id)
         else:
             response_queues[response_queue_id].put((uid, (response_data, status)), block=False)
```

Comment:
it'd be nice to maybe have some kind of zmq wrapper that makes the resulting api comparable to base queues, then you wouldn't need to switch here and could more easily switch out the queue backend (not for this pr, but maybe a follow-up?). Would  make the code easier to read.

Reply:
great idea!!!